### PR TITLE
Upgrade to rails 7

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Review by mjgiarlo — APPROVED** [review]

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -5,8 +5,8 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
-    <%= stylesheet_link_tag "application" %>
-    <%= javascript_include_tag "application", "data-turbo-track": "reload", defer: true %>
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" integrity="sha384-zCbKRCUGaJDkqS1kPbPd7TveP5iyJE0EjAuZQTgFLD2ylzuqKfdKlfG/eSrtxUkn" crossorigin="anonymous">
```
Comment (mjgiarlo) [inline app/views/layouts/application.html.erb:8]: This is a different version of bootstrap than is pinned in the importmap. Is that OK?

Comment (jcoyne) [reply app/views/layouts/application.html.erb:8]: Yeah, it works fine.

File: config/importmap.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# Pin npm packages by running ./bin/importmap
```
Comment (mjgiarlo) [inline config/importmap.rb:3]: When/how often should we be doing this? I guess I'm fuzzy on the new importmap + sprockets combo, mostly in terms of workflow but also in terms of conventions (see prior question about discrepancy between bootstrap js and css). Not blocking the PR, just hoping to understand more before we merge this.

File: package.json [hunk 3]
```diff
@@ -2,17 +2,9 @@
   "name": "technical_metadata",
```
Comment (mjgiarlo) [inline package.json:2]: Do we still need package.json now that most of its contents have been removed?

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (mjgiarlo) [reply app/views/layouts/application.html.erb:8]: Is there any reason to want diff versions of bootstrap? I.e., any reason not to bump the js pin to 4.6.1?

File: package.json (hunk 3 above)
Comment (jcoyne) [reply package.json:2]: I don't think so, but it's possible someone will need something in the future. How about we follow up after to pursue this.

File: yarn.lock [hunk 4]
```diff
@@ -2,251 +2,3 @@
 # yarn lockfile v1
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply yarn.lock:2]: I don't think so, but let's pursue this separately.

File: config/importmap.rb (hunk 2 above)
Comment (jcoyne) [reply config/importmap.rb:3]: you do this when you add a JS dependency. This is the alternative to bunding the js using node. It transfers this work to the client, which makes the app simpler to deploy.

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (jcoyne) [reply app/views/layouts/application.html.erb:8]: As a follow up I was going to upgrade to 5.0. But no, there's no reason to have different versions.

Comment (mjgiarlo) [reply app/views/layouts/application.html.erb:8]: OK, no bigs.

**Pull request merged**